src/main/java/com/celements/course/service/CourseService.java

```diff
@@ -50,11 +87,9 @@ public DocumentReference getCourseTypeForCourse(DocumentReference courseDocRef
   }
 
   @Override
-  public String getCourseTypeName(DocumentReference courseTypeDocRef
-      ) throws XWikiException {
+  public String getCourseTypeName(DocumentReference courseTypeDocRef) throws XWikiException {
     String typeName = "";
-    XWikiDocument typeDoc = getContext().getWiki().getDocument(courseTypeDocRef, 
-        getContext());
+    XWikiDocument typeDoc = getContext().getWiki().getDocument(courseTypeDocRef, getContext());
```

Comment:
please refactor the two lines 92, 93 to one modelAccess.getXObject(docRef, classRef). It ensures that not XWikiDocument is accessible and hence is may not be stored (after readonly get).